android-agent/src/main/java/io/opentelemetry/android/instrumentation/AndroidInstrumentationRegistryImpl.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation
+
+import java.util.Collections
+import java.util.ServiceLoader
+
+internal class AndroidInstrumentationRegistryImpl : AndroidInstrumentationRegistry {
+    private val instrumentations: MutableMap<Class<out AndroidInstrumentation>, AndroidInstrumentation> by lazy {
+        ServiceLoader.load(AndroidInstrumentation::class.java).associateBy { it.javaClass }
+            .toMutableMap()
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : AndroidInstrumentation> get(type: Class<out T>): T? {
+        return instrumentations[type] as? T
+    }
+
+    override fun getAll(): Collection<AndroidInstrumentation> {
+        return Collections.unmodifiableCollection(instrumentations.values)
+    }
```

Comment:
Any reason to use the Java `Collection` types instead of the Kotlin ones?

Reply:
Do you mean `kotlin.collections.Collection`?

Reply:
yep, the `kotlin.collections` package in general.
The less we depend on `java.*` the easier it'll be to be KMP compatible in the future

Reply:
Good point. I was looking for a `Collections.unmodifiableCollection` equivalent, though it seems like `toList` should do. I've just made the changes. Cheers!